### ISSUE TITLE
Download-BcNuGetPackageToFolder : You cannot call a method on a null-valued expression.

### ISSUE THREAD

**Issue by Arthurvdv** [microsoft/navcontainerhelper#3304]
**Describe the issue**
When using the Download-BcNuGetPackageToFolder command I run into an error. Not sure if I'm missing to set some parameter?

BcContainerHelper version 6.0.4

**Scripts used to create container and cause the issue**
```
$bcContainerHelperConfig.TrustedNuGetFeeds = @(
    [PSCustomObject]@{
        "url" = "https://pkgs.dev.azure.com/<organization>/_packaging/<feed>/nuget/v3/index.json";
        "token" = '<redacted>'
    }
    [PSCustomObject]@{
        "url" = "https://dynamicssmb2.pkgs.visualstudio.com/DynamicsBCPublicFeeds/_packaging/MSApps/nuget/v3/index.json"
    }
    [PSCustomObject]@{
        "url" = "https://dynamicssmb2.pkgs.visualstudio.com/DynamicsBCPublicFeeds/_packaging/MSSymbols/nuget/v3/index.json"
    }
)
Download-BcNuGetPackageToFolder -packageName '22b42e4f-a408-402e-90a1-2ebe2f331a59' -folder "C:\<MyUserFolder>" -Verbose
```

**Full output of scripts**
```
Looking for NuGet package 22b42e4f-a408-402e-90a1-2ebe2f331a59 version 0.0.0.0 (Latest match)
Search NuGetFeed https://pkgs.dev.azure.com/<organization>/_packaging/<feed>/nuget/v3/index.json
PackageId: VanRoeyEuroSys.<appName>.22b42e4f-a408-402e-90a1-2ebe2f331a59
Search NuGetFeed https://dynamicssmb2.pkgs.visualstudio.com/DynamicsBCPublicFeeds/_packaging/MSApps/nuget/v3/index.json
Search NuGetFeed https://dynamicssmb2.pkgs.visualstudio.com/DynamicsBCPublicFeeds/_packaging/MSSymbols/nuget/v3/index.json
Best match for package name 22b42e4f-a408-402e-90a1-2ebe2f331a59 Version 0.0.0.0: VanRoeyEuroSys.<appName>.22b42e4f-a408-402e-90a1-2ebe2f331a59 Version 23.2.123209 from https://pkgs.dev.azure.com/<organization>/_packaging/<feed>/nuget/v3/index.json
VERBOSE: NUSPEC:
VERBOSE: <?xml version="1.0" encoding="utf-8"?>
VERBOSE: <package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
VERBOSE:     <metadata>
VERBOSE:         <id>VanRoeyEuroSys.<appName>.22b42e4f-a408-402e-90a1-2ebe2f331a59</id>
VERBOSE:         <version>23.2.123209.0</version>
VERBOSE:         <title><appName></title>
VERBOSE:         <description><appName></description>
VERBOSE:         <authors>VanRoey | EuroSys</authors>
VERBOSE:         <dependencies>
VERBOSE:             <dependency id="Microsoft._Exclude_APIV2_.10cb69d9-bc8a-4d27-970a-9e110e9db2a5" version="19.0.0.0" />
VERBOSE:             <dependency id="Microsoft.Application" version="23.2.0.0" />
VERBOSE:             <dependency id="Microsoft.Platform" version="23.0.0.0" />
VERBOSE:         </dependencies>
VERBOSE:     </metadata>
VERBOSE:     <files>
VERBOSE:         <file src="7836e2f3-df84-40cb-a739-2429918fd3b9.app" target="7836e2f3-df84-40cb-a739-2429918fd3b9.app" />
VERBOSE:     </files>
VERBOSE: </package>
Download-BcNuGetPackageToFolder : You cannot call a method on a null-valued expression.
At line:16 char:1
+ Download-BcNuGetPackageToFolder -packageName '22b42e4f-a408-402e-90a1 ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Download-BcNuGetPackageToFolder], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull,Download-BcNuGetPackageToFolder

```

**Comment by freddydk** [comment]
You shouldn't need any parameters - unfortunately, the error doesn't give me a stack trace:-(   (will try to add that)
and I will try to repro the problem here.

**Comment by freddydk** [comment]
I cannot reproduce the problem here - will make sure that the error in the command provides a better stack trace for me to troubleshoot

**Comment by Arthurvdv** [comment]
Thanks for looking into this! I'll wait for a new release to retry this scenario and report back.

**Comment by freddydk** [comment]
function should now display more info in 6.0.5 - please retry and give me the stacktrace and more.
Thanks

**Comment by Arthurvdv** [comment]
I've updated the BcContainerHelper  and did a retry:

```
BcContainerHelper version 6.0.5
BC.HelperFunctions emits usage statistics telemetry to Microsoft
WARNING: The names of some imported commands from the module 'BcContainerHelper' include unapproved verbs that might make them less discoverable. To find the commands with unapproved verbs, run the Import
-Module command again with the Verbose parameter. For a list of approved verbs, type Get-Verb.

PS C:\windows\system32> 
$bcContainerHelperConfig.TrustedNuGetFeeds = @(
    [PSCustomObject]@{
        "url"   = "https://pkgs.dev.azure.com/vanroeybe/_packaging/BusinessCentral/nuget/v3/index.json";
        "token" = '<redacted>'
    }
    [PSCustomObject]@{
        "url" = "https://dynamicssmb2.pkgs.visualstudio.com/DynamicsBCPublicFeeds/_packaging/MSApps/nuget/v3/index.json"
    }
    [PSCustomObject]@{
        "url" = "https://dynamicssmb2.pkgs.visualstudio.com/DynamicsBCPublicFeeds/_packaging/MSSymbols/nuget/v3/index.json"
    }
)
Download-BcNuGetPackageToFolder -packageName '22b42e4f-a408-402e-90a1-2ebe2f331a59' -folder "C:\Users\arthur.vandevondervo\Downloads" -Verbose
Looking for NuGet package 22b42e4f-a408-402e-90a1-2ebe2f331a59 version 0.0.0.0 (Latest match)
Search NuGetFeed https://pkgs.dev.azure.com/vanroeybe/_packaging/BusinessCentral/nuget/v3/index.json
VERBOSE: GET with 0-byte payload
VERBOSE: received 1696-byte response of content type application/json; charset=utf-8; api-version=7.2-preview.1
VERBOSE: Capabilities of NuGet server https://pkgs.dev.azure.com/vanroeybe/_packaging/BusinessCentral/nuget/v3/index.json are:
VERBOSE: - SearchQueryService=https://pkgs.dev.azure.com/vanroeybe/_packaging/82099a1a-fe04-4c03-9594-3ab7937dc1df/nuget/v3/query2/
VERBOSE: - PackagePublish=https://pkgs.dev.azure.com/vanroeybe/_packaging/82099a1a-fe04-4c03-9594-3ab7937dc1df/nuget/v2/
VERBOSE: - PackageBaseAddress=https://pkgs.dev.azure.com/vanroeybe/_packaging/82099a1a-fe04-4c03-9594-3ab7937dc1df/nuget/v3/flat2/
Search package using https://pkgs.dev.azure.com/vanroeybe/_packaging/82099a1a-fe04-4c03-9594-3ab7937dc1df/nuget/v3/query2/?q=22b42e4f-a408-402e-90a1-2ebe2f331a59
VERBOSE: GET with 0-byte payload
VERBOSE: received 1130-byte response of content type application/json; charset=utf-8; api-version=7.2-preview.1
1 matching packages found
- VanRoeyEuroSys.appname.22b42e4f-a408-402e-90a1-2ebe2f331a59
PackageId: VanRoeyEuroSys.appname.22b42e4f-a408-402e-90a1-2ebe2f331a59
Get versions using https://pkgs.dev.azure.com/vanroeybe/_packaging/82099a1a-fe04-4c03-9594-3ab7937dc1df/nuget/v3/flat2/vanroeyeurosys.appname.22b42e4f-a408-402e-90a1-2ebe2f331a59/index.json
VERBOSE: GET with 0-byte payload
VERBOSE: received 84-byte response of content type application/json; charset=utf-8; api-version=7.2-preview.1
5 versions found
First version is 23.2.123209
Last version is 23.2.122953
Latest version is 23.2.123209
Search NuGetFeed https://dynamicssmb2.pkgs.visualstudio.com/DynamicsBCPublicFeeds/_packaging/MSApps/nuget/v3/index.json
VERBOSE: GET with 0-byte payload
VERBOSE: received 2104-byte response of content type application/json; charset=utf-8; api-version=7.2-preview.1
VERBOSE: Capabilities of NuGet server https://dynamicssmb2.pkgs.visualstudio.com/DynamicsBCPublicFeeds/_packaging/MSApps/nuget/v3/index.json are:
VERBOSE: - SearchQueryService=https://dynamicssmb2.pkgs.visualstudio.com/571e802d-b44b-45fc-bd41-4cfddec73b44/_packaging/d40c2cb2-02c8-4823-8f3b-7c7018269af3/nuget/v3/query2/
VERBOSE: - PackagePublish=https://dynamicssmb2.pkgs.visualstudio.com/571e802d-b44b-45fc-bd41-4cfddec73b44/_packaging/d40c2cb2-02c8-4823-8f3b-7c7018269af3/nuget/v2/
VERBOSE: - PackageBaseAddress=https://dynamicssmb2.pkgs.visualstudio.com/571e802d-b44b-45fc-bd41-4cfddec73b44/_packaging/d40c2cb2-02c8-4823-8f3b-7c7018269af3/nuget/v3/flat2/
Search package using https://dynamicssmb2.pkgs.visualstudio.com/571e802d-b44b-45fc-bd41-4cfddec73b44/_packaging/d40c2cb2-02c8-4823-8f3b-7c7018269af3/nuget/v3/query2/?q=22b42e4f-a408-402e-90a1-2ebe2f331a59

VERBOSE: GET with 0-byte payload
VERBOSE: received 260-byte response of content type application/json; charset=utf-8; api-version=7.2-preview.1
0 matching packages found
Search NuGetFeed https://dynamicssmb2.pkgs.visualstudio.com/DynamicsBCPublicFeeds/_packaging/MSSymbols/nuget/v3/index.json
VERBOSE: GET with 0-byte payload
VERBOSE: received 2110-byte response of content type application/json; charset=utf-8; api-version=7.2-preview.1
VERBOSE: Capabilities of NuGet server https://dynamicssmb2.pkgs.visualstudio.com/DynamicsBCPublicFeeds/_packaging/MSSymbols/nuget/v3/index.json are:
VERBOSE: - SearchQueryService=https://dynamicssmb2.pkgs.visualstudio.com/571e802d-b44b-45fc-bd41-4cfddec73b44/_packaging/fb27d05d-70be-49b0-8e3a-dec41527293c/nuget/v3/query2/
VERBOSE: - PackagePublish=https://dynamicssmb2.pkgs.visualstudio.com/571e802d-b44b-45fc-bd41-4cfddec73b44/_packaging/fb27d05d-70be-49b0-8e3a-dec41527293c/nuget/v2/
VERBOSE: - PackageBaseAddress=https://dynamicssmb2.pkgs.visualstudio.com/571e802d-b44b-45fc-bd41-4cfddec73b44/_packaging/fb27d05d-70be-49b0-8e3a-dec41527293c/nuget/v3/flat2/
Search package using https://dynamicssmb2.pkgs.visualstudio.com/571e802d-b44b-45fc-bd41-4cfddec73b44/_packaging/fb27d05d-70be-49b0-8e3a-dec41527293c/nuget/v3/query2/?q=22b42e4f-a408-402e-90a1-2ebe2f331a59

VERBOSE: GET with 0-byte payload
VERBOSE: received 260-byte response of content type application/json; charset=utf-8; api-version=7.2-preview.1
0 matching packages found
Best match for package name 22b42e4f-a408-402e-90a1-2ebe2f331a59 Version 0.0.0.0: VanRoeyEuroSys.appname.22b42e4f-a408-402e-90a1-2ebe2f331a59 Version 23.2.123209 from https://pkgs.dev.azure.com/vanroe
ybe/_packaging/BusinessCentral/nuget/v3/index.json
Download package using https://pkgs.dev.azure.com/vanroeybe/_packaging/82099a1a-fe04-4c03-9594-3ab7937dc1df/nuget/v3/flat2/vanroeyeurosys.appname.22b42e4f-a408-402e-90a1-2ebe2f331a59/23.2.123209/vanro
eyeurosys.appname.22b42e4f-a408-402e-90a1-2ebe2f331a59.23.2.123209.nupkg
VERBOSE: GET with 0-byte payload
VERBOSE: received 723880-byte response of content type application/octet-stream
Package successfully downloaded
VERBOSE: NUSPEC:
VERBOSE: <?xml version="1.0" encoding="utf-8"?>
VERBOSE: <package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
VERBOSE:     <metadata>
VERBOSE:         <id>VanRoeyEuroSys.appname.22b42e4f-a408-402e-90a1-2ebe2f331a59</id>
VERBOSE:         <version>23.2.123209.0</version>
VERBOSE:         <title>App Name</title>
VERBOSE:         <description>App Name</description>
VERBOSE:         <authors>VanRoey | EuroSys</authors>
VERBOSE:         <dependencies>
VERBOSE:             <dependency id="Microsoft._Exclude_APIV2_.10cb69d9-bc8a-4d27-970a-9e110e9db2a5" version="19.0.0.0" />
VERBOSE:             <dependency id="Microsoft.Application" version="23.2.0.0" />
VERBOSE:             <dependency id="Microsoft.Platform" version="23.0.0.0" />
VERBOSE:         </dependencies>
VERBOSE:     </metadata>
VERBOSE:     <files>
VERBOSE:         <file src="7836e2f3-df84-40cb-a739-2429918fd3b9.app" target="7836e2f3-df84-40cb-a739-2429918fd3b9.app" />
VERBOSE:     </files>
VERBOSE: </package>
Error Message: You cannot call a method on a null-valued expression.
StackTrace: at Download-BcNuGetPackageToFolder, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\6.0.5\NuGet\Download-BcNuGetPackageToFolder.ps1: line 150
at <ScriptBlock>, <No file>: line 15
You cannot call a method on a null-valued expression.
At C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\6.0.5\NuGet\Download-BcNuGetPackageToFolder.ps1:150 char:29
+ ...                         $dependencyCountry = $matches[3].TrimEnd('.')
+                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : InvokeMethodOnNull
```

**Comment by freddydk** [comment]
That looks like a bug

**Comment by Arthurvdv** [comment]
This is resolved in version 6.0.6 and is now working as expected, thank you!
